### ISSUE TITLE
configs are hard to get working

### ISSUE THREAD

**Issue by nicks** [windmilleng/tilt#3191]
From the #tilt channel:

> I'm using `tilt up -- --to-run frontend` and then `config.define_string_list("to-run", args=True)` and `print(cfg.get('to-run', []))` and I get `Error: flag provided but not defined: -to-run`

When I read the documentation myself, I was not able to get this to work, or adequately explain why it didn't work. (The problem is that `args=True` means "don't use the flag name on the command-line")

As I see it, there are a couple different problems here that might be worth separating:
- The error message is poor. Tilt knows that `to-run` is a positional argument. Why can't it tell me that?
- The documentation doesn't link to any working examples. When I tried to make a working example myself, I was not able to. Multiple people have complained that they were not able to get this working without hand-holding from the Tilt team

**Comment by crhayes** [comment]
I've run into similar issues while consulting the docs and implementing custom configuration. There are two issues that I have encountered. I am using Tilt version `v0.13.3`.

1) Docs seem to be incorrect
2) API feels inconsistent

**Docs seem to be incorrect**
For example, the `Specify services to edit` example (https://docs.tilt.dev/tiltfile_config.html) specifies that you should be able to run the command `tilt up -- consumer --to-edit b --to-edit c` to boot up all the `consumer` services in "prod" mode, and services `b` and `c` in "edit" mode.

However, what I get when I implement the associated code example and run `tilt up -- consumer --to-edit b --to-edit c` is:
```starlark
print(cfg.get('to-run', [])); # ["consumer", "--to-edit", "b", "--to-edit", "c"]
print(resources); # ["a", "b", "c", "--to-edit", "b", "--to-edit", "c"]
print(cfg.get('to-edit', [])); # []
```
This works correctly when run as `tilt up -- --to-edit b --to-edit c consumer`;

I have a repro here: https://github.com/crhayes/tilt-config-docs-repro

**API feels inconsistent**
Given a simplified `Tiltfile`:
```starlark
config.define_string_list("to-run", args=True)
config.define_string_list("to-edit")
cfg = config.parse()

print(cfg.get('to-run', []));
print(cfg.get('to-edit', []));

local_resource('a', 'echo "Hi"');
local_resource('b', 'echo "Hi"');
local_resource('c', 'echo "Hi"');
local_resource('d', 'echo "Hi"');
```
Here is a table of commands run and the output of the two print statements:
| Command                                | to-run                       | to-edit |
| -------------------------------------- | ---------------------------- | ------- |
| `tilt up`                              | []                           | []      |
| `tilt up a b`                          | ["a", "b"]                   | []      |
| `tilt up --verbose a b`                | ["a", "b"]                   | []      |
| `tilt up a b --verbose`                | ["a", "b"]                   | []      |
| `tilt up a b -- --to-edit c`           | ["a", "b", "--to-edit", "c"] | []      |
| `tilt up -- a b --to-edit c`           | ["a", "b", "--to-edit", "c"] | []      |
| `tilt up -- --to-edit c a b`           | ["a", "b"]                   | ["c"]   |
| `tilt up -- --to-edit c a b --verbose` | ["a", "b", "--verbose"]      | ["c"]   |
| `tilt up --verbose -- --to-edit c a b` | ["a", "b"]                   | ["c"]   |

In the following I refer to official Tilt API args as `tilt args` and custom ones implemented by us developers as `custom args`.

What I find inconsistent is that, given the exact same `Tiltfile`,  `tilt args` and `positional args` have no required order at the command line so long as `custom args` are **not** specified. Once `custom args` are specified, the location of `positional args` moves to the very end of the command, and the order of `positional args` and `tilt args` is no longer interchangeable.

I guess the argument could be made that `positional args` always come after `custom args`, even when no `custom args` are given, but then it's confusing that in only one of the two cases the order of `tilt args` and `positional args` can be interchanged.

TLDR: What tripped me up the most and is not well documented is the fact that the location of `positional args` moves depending on whether or not `custom args` are specified.
